src/peft/tuners/lora/layer.py

```diff
@@ -181,9 +184,13 @@ def dora_init(self, adapter_name: str) -> None:
         lora_B = self.lora_B[adapter_name]
         scaling = self.scaling[adapter_name]
         with gather_params_ctx(self.get_base_layer()):
-            weight = self.get_base_layer().weight
-            quant_state = getattr(self.get_base_layer(), "state", None)
-            weight = dequantize_bnb_weight(weight, state=quant_state)  # no-op if not bnb
+            base_layer = self.get_base_layer()
+            if hasattr(base_layer, "W_q"):
+                weight = base_layer.dequantize()
+            else:
+                weight = base_layer.weight
+                quant_state = getattr(base_layer, "state", None)
+                weight = dequantize_bnb_weight(weight, state=quant_state)  # no-op if not bnb
```

Comment:
```suggestion
            if hasattr(base_layer, "W_q"):
                weight = base_layer.dequantize()
            else:
                weight = base_layer.weight
                quant_state = getattr(base_layer, "state", None)
                weight = dequantize_bnb_weight(weight, state=quant_state)  # no-op if not bnb
```
Can you add a comment here explaining that this logic is used to take care of quantized linear layers?